controllers/progressiverollout_controller.go

```diff
@@ -92,11 +94,21 @@ func (r *ProgressiveRolloutReconciler) Reconcile(ctx context.Context, req ctrl.R
 		}
 
 		// Get the Applications to update
-		scheduledApps := scheduler.Scheduler(apps, stage)
+		scheduledApps := scheduler.Scheduler(outOfSyncApps, stage)
 
 		for _, s := range scheduledApps {
-			// TODO: add sync method here
 			r.Log.Info("syncing app", "app", s)
+
+			_, err := r.syncApp(s.Name)
+
+			if err != nil {
+				if strings.Contains(err.Error(), "another operation is already in progress") {
+					log.Info("another operation is already in progress", "app", s)
+				} else {
+					log.Error(err, "unable to sync app", "app", s)
+					return ctrl.Result{}, err
```

Comment:
Is this all we want to do? Do we want to somehow signal the stage as failed? Or is that future work?

Reply:
> Do we want to somehow signal the stage as failed

For future work in https://github.com/Skyscanner/argocd-progressive-rollout/issues/30 - as we don't do it at the moment

Reply:
What @maruina said... happy path only :-)